### DISCUSSION TITLE
Adding requeried field with default value triggers data loss in Postgres

### DISCUSSION THREAD

**Discussion by peterHakio** [prisma/prisma#21979] [Q&A]
### Bug description

Adding requeried field with a default value value is completely standard practice in postgres. And doing it should not cause data loss. Postgres has the default behavior to populate the new column with the default value. So it is no problem just to run the migration as is. 

### How to reproduce

Create a model: 
```
model Company {
  orgId String @id
  name String
}

```
Run the migration and a add some data.
Then update the model and add more data:
```
model Company {
  orgId String @id
  currency String
  leadtimeDays Int @default(35)
}
```

results in a migration:

`ALTER TABLE "Company" ADD COLUMN     "leadtimeDays" INTEGER NOT NULL DEFAULT 35;
`

When I run this migration:
```

Datasource "db": PostgreSQL database "test", schema "public" at "localhost:5432"

- Drift detected: Your database schema is not in sync with your migration history.

The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

[*] Changed the `Company` table
  [+] Added column `leadtimeDays`

- The migrations recorded in the database diverge from the local migrations directory. Last common migration: `20231114095020_add_company`. Migrations applied to the database but absent from the migrations directory are: 20231115131733_leadtime_days

✔ We need to reset the "public" schema at "localhost:5432"
Do you want to continue? All data will be lost. … yes
```

### Expected behavior

Prisma should not delete the database when the migration runs. Just run the generated code. It works.

### Prisma information

"@prisma/client": "^5.4.2"

postgres 15

```
generator client {
    provider = "prisma-client-js"
    previewFeatures = ["views"]
}

datasource db {
    provider = "postgresql"
    // NOTE: When using mysql or sqlserver, uncomment the @db.Text annotations in model Account below
    // Further reading:
    // https://next-auth.js.org/adapters/prisma#create-the-prisma-schema
    // https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#string
    url      = env("DATABASE_URL")
}
model Company {
  orgId String @id
  currency String
  leadtimeDays Int @default(35)
}
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by janpio** [comment]
The migration itself is not doing that. Notice these 2 messages from the output:

> - Drift detected: Your database schema is not in sync with your migration history.
> ...
> - The migrations recorded in the database diverge from the local migrations directory. Last common migration: `20231114095020_add_company`. Migrations applied to the database but absent from the migrations directory are: 20231115131733_leadtime_days

That is the problem being detected as drift, and requires a reset to be put in a good state again.

Can you describe exactly which steps (with commands and changes) you did from an empty project?
Can you reproduce the situation and show it to us?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
